scripts/audit-plugin.sh

```diff
@@ -0,0 +1,101 @@
+# <Priority (P1 or P?)> <COMMIT-HASH> <COMMIT-MSG>
+# in this the COMMIT-HASH and COMMIT-MSG are the original sha1 and commit message from spark commit
+
+
+set -ex
+
+nvidia-smi
+
+#Set env variables
+. jenkins/version-def.sh
+
+#Get plugin jar
+ARTF_ROOT="$WORKSPACE/jars"
+MVN_GET_CMD="mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -B \
+    -Dmaven.repo.local=$WORKSPACE/.m2 \
+    $MVN_URM_MIRROR -Ddest=$ARTF_ROOT"
+
+rm -rf $ARTF_ROOT && mkdir -p $ARTF_ROOT
+# maven download SNAPSHOT jars: rapids-4-spark, spark3.0
+$MVN_GET_CMD -DremoteRepositories=$PROJECT_REPO \
+    -DgroupId=com.nvidia -DartifactId=rapids-4-spark_$SCALA_BINARY_VER -Dversion=$PROJECT_VER
+
+RAPIDS_PLUGIN_JAR="$ARTF_ROOT/rapids-4-spark_${SCALA_BINARY_VER}-$PROJECT_VER.jar"
+# Use jdeps to find the dependencies of the rapids-4-spark jar
+DEPS=$(jdeps -include com\.nvidia\.spark.* -verbose:class -e org.apache.spark.* $RAPIDS_PLUGIN_JAR)
+
+FOUND=1
+cd ${SPARK_HOME}
+PRIORITIZED_COMMITS=${WORKSPACE}/commits-with-priority.txt
+if [ -e ${PRIORITIZED_COMMITS} ]; then
+  rm ${PRIORITIZED_COMMITS}
+fi  
+# Read each commit and commit message
+while read COMMIT_ID COMMIT_MSG
+do
+  echo checking commit $COMMIT_ID
+  # Get the scala and java files changed in this commit
+  set +ex
+  GIT_NAME_ONLY=$(git show $COMMIT_ID --name-only --oneline | grep '.*scala\|java')
+  set -ex
+  CLASSES=""
+  for i in $GIT_NAME_ONLY
+  do
+    # Get the classes from the each file. This is tricky and is only relying on picking out the 
+    # object/trait/class from the git show command. It then tries to figure out the fully-qualified 
+    # class name from the file name and store them in LOCAL_CLASSES. We can make this regex better to 
+    # focus on only code changes.
+    set +ex
+    LOCAL_CLASSES=$(git show $COMMIT_ID -- $i | grep '\sclass\s\|\sobject\s\|\strait\s'|sed 's/.*trait \([[:alnum:]]*\).*/\1/' | sed 's/.*class \([[:alnum:]]*\).*/\1/' | sed 's/.*object \([[:alnum:]]*\).*/\1/')
+    set -ex
+    # Figure out the package from the file name
+    PACKAGE=$(dirname -- $i | sed 's/.*scala\///g' | sed 's/.*java\///g' | sed 's/\//\./g')
+    for CLASS in $LOCAL_CLASSES
+    do
+      # Prepend the package to the class name
+      CLASSES="${CLASSES}\n$PACKAGE.${CLASS}"
+    done
+  done
+  # dedupe
+  for CLASS in $(echo -e $CLASSES|sort|uniq|xargs)
+  do
+  if [ ! -z "${CLASS}" ]; then
+    # Look for the class name in the dependencies to determine if this change affects us
+    set +ex
+    FOUND=$(grep -c ${CLASS} <<< ${DEPS})
+    set -ex
+    echo -e "looking for class ${CLASS} and ${FOUND}\n"
+    if [ ${FOUND} -ne 0 ]; then 
+      printf "%s\t%s\t%s\t%s\n" "P1" "${COMMIT_ID}" "${COMMIT_MSG}" "${CLASS}" >> ${PRIORITIZED_COMMITS}
+      break
+    fi
+  fi
+  done
+  if [ ${FOUND} -eq 0 ]; then
+    printf "%s\t%s\t%s\n" "P?" "${COMMIT_ID}" "${COMMIT_MSG}" >> ${PRIORITIZED_COMMITS}
+  fi
+done < <(cat ${WORKSPACE}/hashCommitsWithMessage.log)
```

Comment:
hashCommitsWithMessage.log is used to get the initial set of commits and not for incremental set of changes. I think we  should use $COMMIT_DIFF_LOG instead.

Reply:
is that an env variable expected by `audit-spark-3.2.sh` because I don't see it being set in that script

Reply:
Also, what about the first time we run the script? We need to get both initial and subsequent runs.

Reply:
Yes, it's environment variable setup in jenkins job. 
Thanks for pointing it out. I think we could write to $COMMIT_DIFF_LOG for both the cases and read that file in this script.